frontend/src/logic/real-schedule-logic/schedule.logic.ts

```diff
@@ -6,40 +6,38 @@ import { ChildrenInfoLogic } from "./children-info.logic";
 import { DataRow } from "./data-row";
 import { MetadataLogic } from "./metadata.logic";
 import { ShiftsInfoLogic } from "./shifts-info.logic";
+import { ScheduleProvider, Schedule } from "../schedule-provider";
 
-interface ScheduleSections {
-  nurseInfo: ShiftsInfoLogic,
-  babysitterInfo: ShiftsInfoLogic,
-  childrenInfo: ChildrenInfoLogic
-}
+export class ScheduleLogic implements ScheduleProvider {
+  nurseInfoProvider: ShiftsInfoLogic;
+  babysitterInfoProvider: ShiftsInfoLogic;
+  childrenInfoProvider: ChildrenInfoLogic;
 
-export class ScheduleLogic {
-  private sections: ScheduleSections;
```

Comment:
Pozbyłem się sekcji, żeby być zachować spójność z tym co jest w Parserze. Poszedłem w tę stronę trochę z lenistwa, bo nie chciałem za bardzo ingerować w kod Parsera. Druga kwestia to to, że sekcji jest niewiele, i dodatkowo ich obsługa różni się w zależności od typu. Chodzi mi o funkcji updateSection, która obecnie została rozdzielona na 3 oddzielne funkcje co moim zdaniem zmniejsza ryzko pomyłki + nie mamy sprawdzenia ```section === "childrenInfo"```, które jest dość niebezpieczne.

Reply:
Brzmi sensownie. Zgadzam się z tobą że w tym przypadku nie musimy dążyć do tego, żeby wszystko było bardzo generyczne i podział wszystkiego na oddzielne metody jest sensowne